scripts/list_boards.py

```diff
@@ -232,8 +279,23 @@ def load_v2_boards(board_name, board_yml, systems):
                 variants=[Variant.from_dict(v) for v in board.get('variants', [])],
                 hwm='v2',
             )
-            boards.append(board)
-    return boards
+    return boards, board_extensions
+
+
+def extend_v2_boards(boards, board_extensions):
+    for e in board_extensions:
+        board = boards.get(e['extend'])
+        if board is None:
+            continue
+        board.directories.append(e['dir'])
+
+        for v in e.get('variants', []):
+            node = board.from_qualifier(v['qualifier'])
+            if str(v['qualifier'] + '/' + v['name']) in board_v2_qualifiers(board):
+                board_yml = e['dir'] / BOARD_YML
+                sys.exit(f'ERROR: Variant: {v["name"]}, defined multiple times for board: '
+                         f'{board.name}.\nLast defined in {board_yml}')
```

Comment:
This check should be made more general. It catches duplicate qualifiers added by extensions, but doesn't catch duplicate qualifiers in the main board definition:
```yaml
board:
  name: nrf54h20dk
  socs:
  - name: nrf54h20
    variants:
    - name: xip
      cpucluster: cpuppr
    - name: xip
      cpucluster: cpuppr
```

Reply:
> but doesn't catch duplicate qualifiers in the main board definition:

True, and I will take a look.
But in practice, a duplicate entry in the main board definition is not an issue in practice because the normalized board target will still pick up the same files from the same board folder, in your example `<main_board_dir>/nrf54h20dk_nrf54h20_cpuppr_xip_defconfig`.

The real issue with duplicate / colliding qualifiers from main board vs extended board is related to ambiguity on where a board variant is really defined, meaning from where will it be picked up.

Reply:
Added a qualifier post check when creating a new Board instance.